tests/components/roomba/test_config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""Test the iRobot Roomba config flow."""
+from asynctest import MagicMock, PropertyMock, patch
+from roomba import RoombaConnectionError
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.roomba.const import (
+    CONF_BLID,
+    CONF_CERT,
+    CONF_CONTINUOUS,
+    CONF_DELAY,
+    DOMAIN,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+
+from tests.common import MockConfigEntry
+
+VALID_CONFIG = {CONF_HOST: "1.2.3.4", CONF_BLID: "blid", CONF_PASSWORD: "password"}
+
+VALID_YAML_CONFIG = {
+    CONF_HOST: "1.2.3.4",
+    CONF_BLID: "blid",
+    CONF_PASSWORD: "password",
+    CONF_CERT: "/etc/ssl/certs/ca-certificates.crt",
+    CONF_CONTINUOUS: True,
+    CONF_DELAY: 1,
+}
+
+
+def _create_mocked_roomba(
+    roomba_connected=None, master_state=None, connect=None, disconnect=None
+):
+    mocked_roomba = MagicMock()
+    type(mocked_roomba).roomba_connected = PropertyMock(return_value=roomba_connected)
+    type(mocked_roomba).master_state = PropertyMock(return_value=master_state)
+    type(mocked_roomba).connect = MagicMock(side_effect=connect)
+    type(mocked_roomba).disconnect = MagicMock(side_effect=disconnect)
+    return mocked_roomba
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5343767)